CI lint fails to run

The [CI lint test](https://ucsc-ci.com/databiosphere/toil/-/jobs/46292) is failing with the errors:
```
src/toil/lib/aws/session.py:40:1: error: Module "botocore.credentials" has no
attribute "JSONFileCache"  [attr-defined]
    from botocore.credentials import JSONFileCache
    ^
src/toil/lib/aws/iam.py:10:1: error: Module "mypy_boto3_iam.type_defs" has no
attribute "AttachedPolicyTypeDef"  [attr-defined]
    from mypy_boto3_iam.type_defs import AttachedPolicyTypeDef
    ^
Found 2 errors in 2 files (checked 67 source files)
```

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-1370)
┆Issue Number: TOIL-1370

--- a/requirements-aws.txt
+++ b/requirements-aws.txt
@@ -1,3 +1,3 @@
 boto>=2.48.0, <3
-boto3-stubs[s3,sdb,iam,sts,boto3]>=1.20.46, <2
+boto3-stubs[s3,sdb,iam,sts,boto3]>=1.28.3, <2
 moto>=4.1.11, <5

--- a/src/toil/lib/aws/iam.py
+++ b/src/toil/lib/aws/iam.py
@@ -7,7 +7,7 @@ from typing import Any, Dict, List, Optional, Set, cast, Union, Sequence
 
 import boto3
 from mypy_boto3_iam import IAMClient
-from mypy_boto3_iam.type_defs import AttachedPolicyTypeDef
+from mypy_boto3_iam.type_defs import AttachedPolicyOutputTypeDef
 from mypy_boto3_sts import STSClient
 
 from toil.lib.aws import zone_to_region
@@ -145,7 +145,7 @@ def get_actions_from_policy_document(policy_doc: Dict[str, Any]) -> AllowedActio
                             allowed_actions[resource][key].append(statement[key])
 
     return allowed_actions
-def allowed_actions_attached(iam: IAMClient, attached_policies: List[AttachedPolicyTypeDef]) -> AllowedActionCollection:
+def allowed_actions_attached(iam: IAMClient, attached_policies: List[AttachedPolicyOutputTypeDef]) -> AllowedActionCollection:
     """
     Go through all attached policy documents and create an AllowedActionCollection representing granted permissions.
 

--- a/src/toil/lib/aws/session.py
+++ b/src/toil/lib/aws/session.py
@@ -37,7 +37,7 @@ import boto.connection
 import botocore
 from boto3 import Session
 from botocore.client import Config
-from botocore.credentials import JSONFileCache
+from botocore.utils import JSONFileCache
 from botocore.session import get_session
 
 logger = logging.getLogger(__name__)

